languageservice/src/validate-action-reference.ts

```diff
@@ -0,0 +1,92 @@
+import {isMapping} from "@actions/workflow-parser";
```

Comment:
This is actually existing code

The file was renamed from `validate-action.ts`

Reply:
(i added new comments throughout the file)